carla_controller/transform_dataset.py

```diff
@@ -0,0 +1,107 @@
+import carla
+import cv2
+import os
+
+import numpy as np
+from pathlib import Path
+from shutil import copy
+from typing import Dict
+
+
+SOURCE_DIRECTORY = "./output"
+DESTINATION_DIRECTORY = "./dataset"
+
+
+def load_dataset_stats():
+    # file_stats is an object 
+    file_stats = {}
+
+    directories = os.listdir(SOURCE_DIRECTORY)
+    for dir in directories:
+        files = os.listdir(f"{SOURCE_DIRECTORY}/{dir}/semantic")
+
+        for file in files:
+            pass
+
+
+# def read_semantic_image(image_path : str) -> Dict:
+#     img = cv2.imread(image_path, cv2.IMREAD_COLOR)
+
+#     stats = {}
+
+
+def convert_semantic_image(image_path : str):
+    img = cv2.imread(image_path, cv2.IMREAD_COLOR)
+
+    for key in SEMANTIC_CATEGORIES.keys():
+        r_channel = (0, 0, key)
+        category = SEMANTIC_CATEGORIES[key]
+        color = SEMANTIC_COLORS[category]
+
+        img[np.all(img == r_channel, axis=-1)] = color
+    
+    img = cv2.cvtColor(img, cv2.COLOR_RGB2BGR)
+
+    return img
+
+
+def transform_dataset():
```

Comment:
quick nit on a docstring.  Might not matter that much for us since most of us wont be in the code base, but took me a  bit to figure out which was converting the labeled images into something "human readable"

Reply:
Fair. I'll make some doc notes.